### PR TITLE
Implementation of the key vault factory with deployment

### PR DESCRIPTION
**Pull request by Ashish0410verma**

### CODE REVIEW

**Comment by amitxagarwal** [comment]
why a separate PR? Please put this change in you other PR where you are putting a layer to handle key vault requests.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -57,6 +57,7 @@ stages:
       clientId: $(McaClientId)
       clientSecret: $(McaClientSecret)
       isDbRequired: $(DbRequired)
+      isKeyVaultRequired: 'true'
```
Comment (amitxagarwal) [inline azure-pipelines.yml:60]: we need this in internal as well right? If yes then why this flag is not a part of Internal stage?

File: deploy/azuredeploy.json [hunk 2]
```diff
@@ -355,11 +396,56 @@
...
+            "objectId": "[reference(resourceId('Microsoft.Web/sites', variables('webAppName')), '2016-08-01', 'Full').identity.principalId]",
+            "permissions": {
+              "keys": [
+                "all"
+              ],
+              "secrets": [
+                "all"
+              ],
+              "certificates": [
+                "all"
```
Comment (amitxagarwal) [inline deploy/azuredeploy.json:440]: why we need this?

File: deploy/azuredeploy.json [hunk 3]
```diff
@@ -355,11 +396,56 @@
...
+          "name": "[parameters('vaultSku')]"
+        },
+        "tenantId": "[reference(resourceId('Microsoft.Web/sites', variables('webAppName')), '2016-08-01', 'Full').identity.tenantid]",
+        "accessPolicies": [
+          {
+            "tenantId": "[reference(resourceId('Microsoft.Web/sites', variables('webAppName')), '2016-08-01', 'Full').identity.tenantid]",
+            "objectId": "[reference(resourceId('Microsoft.Web/sites', variables('webAppName')), '2016-08-01', 'Full').identity.principalId]",
+            "permissions": {
+              "keys": [
+                "all"
```
Comment (amitxagarwal) [inline deploy/azuredeploy.json:434]: please enable permissions to only one type. Not for all three.

File: src/Kmd.Momentum.Mea.Api/Controllers/Health/HealthController.cs [hunk 4]
```diff
@@ -50,5 +53,21 @@ public async Task<IActionResult> Live()
...
+        /// <summary>
+        ///     Get secret
+        /// </summary>
+        /// <remarks>Provides an indication about the health of the API</remarks>
+        /// <response code="200">API is healthy</response>
+        /// <response code="503">API is unhealthy or in degraded state</response>
+        [HttpGet]
+        [ProducesResponseType(typeof(HealthReport), (int)HttpStatusCode.OK)]
+        // [Authorize(Scopes.Access)]
+        public async Task<ActionResult<string>> GetSecret([FromQuery] string secretIdentifier)
```
Comment (amitxagarwal) [inline src/Kmd.Momentum.Mea.Api/Controllers/Health/HealthController.cs:66]: why we need this?

File: azure-pipelines.yml (hunk 1 above)
Comment (Ashish0410verma) [reply azure-pipelines.yml:60]: done

File: deploy/azuredeploy.json (hunk 2 above)
Comment (Ashish0410verma) [reply deploy/azuredeploy.json]: Removed

File: deploy/azuredeploy.json (hunk 3 above)
Comment (Ashish0410verma) [reply deploy/azuredeploy.json]: Ok makes sense

File: src/Kmd.Momentum.Mea.Api/Controllers/Health/HealthController.cs (hunk 4 above)
Comment (Ashish0410verma) [reply src/Kmd.Momentum.Mea.Api/Controllers/Health/HealthController.cs:66]: i would keep this as an endpoint in health controller to test the actual functionality of retrieving secret value from key vault. As right now in code we are not fetching values from key vault which will be done in next PR

Comment (amitxagarwal) [reply src/Kmd.Momentum.Mea.Api/Controllers/Health/HealthController.cs:66]: so then are you gonna remove this in the next PR?

Comment (Ashish0410verma) [reply src/Kmd.Momentum.Mea.Api/Controllers/Health/HealthController.cs:66]: Yes

**Pull request merged**